reporting/test/setup.unit.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * WhoTracks.Me
+ * https://ghostery.com/whotracksme
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import chrome from 'sinon-chrome';
+import * as chai from 'chai';
+import sinonChai from 'sinon-chai';
+import chaiAsPromised from 'chai-as-promised';
+
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+chrome.storage.session = chrome.storage.local;
```

Comment:
IMHO, it is better to have such code scoped in a test. That allows to test feature detection and avoids the problem that tests need to clean up the keys that they persisted in the global object.

We also have a mock implementation of the session API here: reporting/test/helpers/fake-session-storage.js.
It can be used like that:
```
chrome.storage.session = new FakeSessionApi();
```;

This reminds me also of this problem here:
https://github.com/whotracksme/webextension-packages/blob/597d1bc2fd58346ccd46e8312d5a1bcf1e6bec31/
reporting/test/request/steps/oauth-detector.spec.js#L44

Where the anti-tracking tests failed when a promise-based API is being provided. They had their own stub implementation, which slightly different from the real API. Thus, the FakeSessionApi is currently not used for the old anti-tracking tests (only for Human Web tests).

Reply:
this is a quick solution for `sinon-chrome` missing new APIs.

Reply:
As as I understand it has drawbacks. It creates a test environment where the keys that are written by one test carry over to the new tests (because it is a global). And also, it makes chrome.storage and chrome.session being mirrored: if you put a key in the session, it will show up in the storage as well.

The alternative is to have each test be responsible for the setup (and clean up afterwards). That is also what the example in the sinon-chrome documentation does (https://www.npmjs.com/package/sinon-chrome):

```
    before(function () {
        global.chrome = chrome;
        this.events = new EventsModule();
    });

    after(function () {
        chrome.flush();
        delete global.chrome;
    });
```

Maybe better to save the old value and restore it, then to "delete". But the basic idea is to shift the setup into the tests. There is also the "flush" method that is expected to be called.

Reply:
```
 beforeEach(function () {
    chrome.flush();
```
does the job to always start from clean

Reply:
btw, this was the reason why I had to create the WebRequestListenerManager - as listeners attached on module being loaded got removed on flush